examples/peripherals/usb/host/hid/main/hid_host_example.c

```diff
@@ -329,16 +329,19 @@ static void hid_host_mouse_report_callback(const uint8_t *const data, const int
 
     static int x_pos = 0;
     static int y_pos = 0;
+    static int wheel_pos = 0;
 
     // Calculate absolute position from displacement
     x_pos += mouse_report->x_displacement;
     y_pos += mouse_report->y_displacement;
+    wheel_pos += mouse_report->scrollwheel;
```

Comment:
Current version of HID Driver was made for devices which have a HID boot interface. In that case the mouse report is simple and always has the default structure. Structure definition can be found in Appendix B2 of [Device Class Definition for Human Interface Devices (HID) Version 1.11](https://www.usb.org/sites/default/files/documents/hid1_11.pdf).

That is why the the structure in the hid_usage_mouse.h file for hid_mouse_input_report_boot_t satisfy the specification. 
Sure, the scroll data in report for different devices pretty much always on the same place in buffer, but to be sure about the correct way is to handle the report device descriptor and handle the report data according to structure from descriptor.